tests/unit/lms/product/d2l/_plugin/grouping_test.py

```diff
@@ -0,0 +1,123 @@
+            sentinel.grading_student_id,
+            type_=Grouping.Type.D2L_GROUP,
+            group_set_id=group_set_id,
+        )
+        assert api_groups == grouping_service.get_course_groupings_for_user.return_value
+
+    def test_get_groups_for_instructor(
+        self, d2l_api_client, plugin, grouping_service, course
+    ):
+        api_groups = plugin.get_groups_for_instructor(
+            grouping_service, course, sentinel.group_set_id
+        )
+
+        d2l_api_client.group_set_groups.assert_called_once_with(
+            course.lms_id, sentinel.group_set_id
+        )
+        assert api_groups == d2l_api_client.group_set_groups.return_value
+
+    def test_get_groups_for_instructor_raises(
+        self, d2l_api_client, plugin, grouping_service, course
+    ):
+        d2l_api_client.group_set_groups.side_effect = ExternalRequestError(
+            response=Mock(status_code=500)
+        )
+
+        with pytest.raises(ExternalRequestError):
+            plugin.get_groups_for_instructor(
+                grouping_service, course, sentinel.group_set_id
+            )
+
+    def test_get_groups_for_instructor_group_set_not_found(
+        self, d2l_api_client, plugin, grouping_service, course
+    ):
+        d2l_api_client.group_set_groups.side_effect = ExternalRequestError(
+            response=Mock(status_code=404)
+        )
+
+        with pytest.raises(GroupError) as err:
+            plugin.get_groups_for_instructor(
+                grouping_service, course, sentinel.group_set_id
+            )
+        assert err.value.error_code == ErrorCodes.GROUP_SET_NOT_FOUND
+
+    def test_get_groups_for_instructor_group_set_empty(
+        self, d2l_api_client, plugin, grouping_service, course
+    ):
+        d2l_api_client.group_set_groups.return_value = []
+
+        with pytest.raises(GroupError) as err:
+            plugin.get_groups_for_instructor(
+                grouping_service, course, sentinel.group_set_id
+            )
+        assert err.value.error_code == ErrorCodes.GROUP_SET_EMPTY
+
+    @pytest.mark.parametrize(
+        "user_id,api_user_id",
+        [
+            ("shoolname_ID", "ID"),
+            ("shoolname_prod_ID", "ID"),
+            ("a72b0b30-5af8-4408-92a8-bffef472c4a7_ID", "ID"),
+        ],
+    )
+    def test__get_api_user_id(self, user_id, api_user_id):
+        # pylint: disable=protected-access
+        assert D2LGroupingPlugin._get_api_user_id(user_id) == api_user_id
+
+    def test_factory(self, pyramid_request, d2l_api_client):
+
+        plugin = D2LGroupingPlugin.factory(sentinel.context, pyramid_request)
+        assert isinstance(plugin, D2LGroupingPlugin)
+        # pylint: disable=protected-access
+        assert plugin._d2l_api == d2l_api_client
+
+    @pytest.fixture
+    def plugin(self, d2l_api_client):
+        return D2LGroupingPlugin(d2l_api_client, sentinel.api_user_id)
+
+    @pytest.fixture
+    def course(self):
+        return factories.Course()
```

Comment:
:+1: These tests are very boring in the best possible way. Very clear and 1:1 with the code they test. No frills or confusing parts.